aiidalab_widgets_base/process.py

```diff
@@ -52,28 +55,32 @@ def on_click(self, function):
 
     def on_btn_submit_press(self, _):
         """When submit button is pressed."""
-        from IPython.display import clear_output
-        from aiida.engine import submit
-
-        with self.submit_out:
-            clear_output()
+        self.submit_out.value = ''
+        input_dict = self.input_dictionary_function()
+        if input_dict is not None:
             self.btn_submit.disabled = True
-            input_dict = self.widgets_values()
             self.process = submit(self._process_class, **input_dict)
-            print("Submitted process {}".format(self.process))
-            return
+            self.submit_out.value = "Submitted process {}".format(self.process)
+            for func in self._run_after_submitted:
+                func(self.process)
+        else:
+            self.submit_out.value = "SubmitButtonWidget: did not recieve input dictionary."
+
+    def run_after_submitted(self, function):
```

Comment:
I think it would be better to use the standard way of implementing such callbacks here:
```
def _on_submitted(self):
    pass
```
The user can then override this function. If they want to register multiple functions, they can do that by creating a meta-function that calls the other functions.

Reply:
Here I was doing something similar to what is done for the `Button` widget.
```
from ipywidgets import Button
b = Button()
b.on_click(lambda _: print("hello world"))
b.on_click(lambda _: print("test"))
display(b)
```
Don't you like this approach?

Reply:
My bad I did not remember correctly how the `Button.on_click()` is implemented. Please keep it *as-is*.

Reply:
The only thing I might suggest is renaming the function to `_on_submitted` or `on_submitted`.